src/recipe/recipe.service.ts

```diff
@@ -55,6 +70,8 @@ export class RecipeService {
       throw new ForbiddenException();
     }
 
+    recipe.imageKeys = await this.getImageUrlsToUrls(recipe);
+
     return recipe;
```

Comment:
again - check if you can use spread operator here. If `recipe` is an interface then you should be able to, if class object then I don't think so. Also the await might be a problem but I think it should pass.

```suggestion
    return { ...recipe, imageKeys:  await this.getImageUrlsToUrls(recipe) }
```